fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -438,6 +439,28 @@ def no_sync(self) -> Generator:
             for m, old_flag in old_flags:
                 m.require_backward_grad_sync = old_flag
 
+    @contextlib.contextmanager
+    def summon_full_params(self) -> Generator:
```

Comment:
can we use this to replace similar logic in state_dict and load_state_dict too?

Reply:
let me try it. If works, will just make the change and merge it in.